apps/web/pages/settings/organizations/platform/oauth-clients/components/OAuthClientForm.tsx

```diff
@@ -25,27 +25,27 @@ type FormValues = {
   scheduleWrite: boolean;
   appsRead: boolean;
   appsWrite: boolean;
+  profileRead: boolean;
+  profileWrite: boolean;
 };
 
 export const OAuthClientForm: FC = () => {
   const { t } = useLocale();
   const router = useRouter();
-  const { register, handleSubmit, setValue, getValues } = useForm<FormValues>({});
+  const { register, handleSubmit, setValue } = useForm<FormValues>({});
   const [isSelectAllPermissionsChecked, setIsSelectAllPermissionsChecked] = useState(false);
 
-  const selectAllPermissions = () => {
-    const allInputValues = getValues();
+  const selectAllPermissions = useCallback(() => {
+    Object.keys(PERMISSIONS_GROUPED_MAP).forEach((key) => {
```

Comment:
Personally, i would still split the permissions out because now if some permission's "key" is changed in PERMISSIONS_GROUPED_MAP or in FormValues then it would break but your call.